ESLint config not found when missing from application root

And eslint plugin dependency resolution gets funky when adding a `.eslintrc.js` to `/shell`.

(Credit @awgaan)

@varl this seems to be an issue when eslint plugins aren't hoisted up to `node_modules` from `@dhis2/cli-style/node_modules`?

```sh
./node_modules/.bin/d2-style js check
d2-style > javascript 
Cannot find module 'eslint-config-prettier'
Referenced from: /Users/awm/dev/dhis2/frontend/tools/platform/shell/node_modules/
@dhis2/cli-style/config/js/eslint.config.js
Referenced from: /Users/awm/dev/dhis2/frontend/tools/platform/shell/node_modules/
@dhis2/cli-style/config/js/eslint-react.config.js
Referenced from: /Users/awm/dev/dhis2/frontend/tools/platform/shell/.eslintrc.js
➜  shell git:(feat/compiler-2dot0) ✗ ./node_modules/.bin/d2-style --version
5.0.3
➜  shell git:(feat/compiler-2dot0) ✗ ls ./node_modules | grep eslint
@typescript-eslint
eslint
eslint-import-resolver-node
eslint-module-utils
eslint-plugin-react
eslint-scope
eslint-utils
eslint-visitor-keys
➜  shell git:(feat/compiler-2dot0) ✗ ls ./node_modules/@dhis2/cli-style/node_modules | grep eslint
babel-eslint
eslint
eslint-config-prettier
eslint-plugin-eslint-plugin
eslint-plugin-react
```

--- a/shell/.eslintrc.js
+++ b/shell/.eslintrc.js
@@ -1,0 +1,13 @@
+const fs = require('fs'),
+    path = require('path')
+
+const customConfigFile = path.resolve('../../.eslintrc.js')
+const hasCustomConfig = fs.existsSync(customConfigFile)
+
+const extendsArray = hasCustomConfig
+    ? [require(customConfigFile)]
+    : ['react-app']
+
+module.exports = {
+    extends: extendsArray,
+}

